deploy/stacks/pipeline.py

```diff
@@ -894,8 +942,17 @@ def set_albfront_stage(self, target_env, repository_name):
                         f'export internet_facing={target_env.get("internet_facing", False)}',
                         f'export custom_domain=True',
                         f'export deployment_region={target_env.get("region", self.region)}',
-                        f'export enable_cw_rum={target_env.get("enable_cw_rum", False)}',
+                        f'export enable_cw_rum={target_env.get("enable_cw_rum", False) and target_env.get("custom_auth", None) == None}',
                         f'export resource_prefix={self.resource_prefix}',
+                        f'export reauth_ttl={str(target_env.get("reauth_config", {}).get("ttl", 5))}',
+                        f'export custom_auth_provider={str(target_env.get("custom_auth", {}).get("provider", "None"))}',
+                        f'export custom_auth_url={str(target_env.get("custom_auth", {}).get("url", "None"))}',
+                        f'export custom_auth_redirect_url={str(target_env.get("custom_auth", {}).get("redirect_url", "None"))}',
+                        f'export custom_auth_client_id={str(target_env.get("custom_auth", {}).get("client_id", "None"))}',
+                        f'export custom_auth_response_types={str(target_env.get("custom_auth", {}).get("response_types", "None"))}',
+                        f'export custom_auth_scopes={str(target_env.get("custom_auth", {}).get("scopes", "None"))}',
+                        f'export custom_auth_claims_mapping_email={str(target_env.get("custom_auth", {}).get("claims_mapping", {}).get("email", "None"))}',
+                        f'export custom_auth_claims_mapping_user_id={str(target_env.get("custom_auth", {}).get("claims_mapping", {}).get("user_id", "None"))}',
```

Comment:
Lines 956 - 961 causes the CICD Pipelien to fail for VPC Facing deployment with custom auth:

```
        "mkdir ~/.aws/ && touch ~/.aws/config",
        "echo \"[profile buildprofile]\" > ~/.aws/config",
        "echo \"role_arn = arn:aws:iam::ACCOUNT_ID:role/RESOURCE_PREFIX-ENVNAME-cognito-config-role\" >> ~/.aws/config",
        "echo \"credential_source = EcsContainer\" >> ~/.aws/config",
        "aws sts get-caller-identity --profile buildprofile",
        "export AWS_PROFILE=buildprofile",
```

This is because the `...-cognito-config-role` only gets created in `IdPStack()` if `custom_auth` 

This role needs to be created regardless of `custom_auth` or not to allow tooling account codeBuild to assume it and read from SSM in the deployment account in `frontend_config.py` python script

Reply:
Since `...-cognito-config-role` is also used in `cognito_config_action()` for NON custom_auth deployments - it may be best if `custom_auth` instead of deploying the `IdPStack()` create a new role with the same permissions as the `...-cognito-config-role` minus the following:
- Cognito Permissions + Resources
- CW Rum Permissions + Resources
- Secrets Manager Permissions + Resources (used for CW Canaries)
So pretty much a new role in backend with SSM permissions that the tooling account can assume

Reply:
Hi @noah-paige , Thanks for pointing this out. Can you tell if this is the policy you had in mind ?

```python
 actions=[
                    'secretsmanager:DescribeSecret',
                    'secretsmanager:GetSecretValue',
                    'ssm:GetParameterHistory',
                    'ssm:GetParameters',
                    'ssm:GetParameter',
                    'ssm:GetParametersByPath',
                    'kms:Decrypt',
                    'kms:GenerateDataKey',
                    'kms:DescribeKey'
                ],
 resources=[
                    f'arn:aws:ssm:*:{self.account}:parameter/*{resource_prefix}*',
                    f'arn:aws:ssm:*:{self.account}:parameter/*dataall*',
                    f'arn:aws:secretsmanager:{self.region}:{self.account}:secret:*{resource_prefix}*',
                    f'arn:aws:secretsmanager:{self.region}:{self.account}:secret:*dataall* 
                ],
```

Reply:
No KMS permissions should be needed either - just SSM

Reply:
Updated the code. with the policy needed for frontend deploy